plugins/woocommerce-admin/client/products/fields/attribute-control/attribute-control.tsx

```diff
@@ -9,56 +9,76 @@ import {
 	__experimentalSelectControlMenuSlot as SelectControlMenuSlot,
 	Link,
 } from '@woocommerce/components';
-import { recordEvent } from '@woocommerce/tracks';
 import interpolateComponents from '@automattic/interpolate-components';
 import { getAdminLink } from '@woocommerce/settings';
 
 /**
  * Internal dependencies
  */
 import './attribute-field.scss';
-import { AddAttributeModal } from './add-attribute-modal';
 import { EditAttributeModal } from './edit-attribute-modal';
 import { EnhancedProductAttribute } from '~/products/hooks/use-product-attributes';
 import {
+	getAttributeId,
 	getAttributeKey,
 	reorderSortableProductAttributePositions,
 } from './utils';
 import { AttributeEmptyState } from '../attribute-empty-state';
 import {
-	AddAttributeListItem,
 	AttributeListItem,
+	NewAttributeListItem,
 } from '../attribute-list-item';
+import { NewAttributeModal } from './new-attribute-modal';
 
 type AttributeControlProps = {
 	value: ProductAttribute[];
+	onAdd?: ( attribute: EnhancedProductAttribute[] ) => void;
 	onChange: ( value: ProductAttribute[] ) => void;
-	// TODO: should we support an 'any' option to show all attributes?
-	attributeType?: 'regular' | 'for-variations';
+	onEdit?: ( attribute: ProductAttribute ) => void;
+	onRemove?: ( attribute: ProductAttribute ) => void;
+	onRemoveCancel?: ( attribute: ProductAttribute ) => void;
+	onNewModalCancel?: () => void;
+	onNewModalClose?: () => void;
+	onNewModalOpen?: () => void;
+	onEditModalCancel?: ( attribute?: ProductAttribute ) => void;
+	onEditModalClose?: ( attribute?: ProductAttribute ) => void;
+	onEditModalOpen?: ( attribute?: ProductAttribute ) => void;
+	text?: {
+		emptyStateSubtitle?: string;
+		newAttributeListItemLabel?: string;
+		newAttributeModalTitle?: string;
+		globalAttributeHelperMessage: string;
+	};
 };
 
 export const AttributeControl: React.FC< AttributeControlProps > = ( {
 	value,
-	attributeType = 'regular',
+	onAdd = () => {},
 	onChange,
+	onEdit = () => {},
+	onNewModalCancel = () => {},
+	onNewModalClose = () => {},
+	onNewModalOpen = () => {},
+	onEditModalCancel = () => {},
+	onEditModalClose = () => {},
+	onEditModalOpen = () => {},
+	onRemove = () => {},
+	onRemoveCancel = () => {},
```

Comment:
Setting default values for these props doesn't really prevent us from ideally checking that the prop is a function before trying to call it, if we want our code to be robust. I've changed my opinion on the setting of default no-op functions for props because of this. Instead, I think we should do checks such as `typeof onAdd === 'function'` before trying to call the prop.

Reply:
As discussed via Slack, we'll keep this as no-ops given it allows us to more quickly identify errors.